Location data twillio channel (#12162)
RasaHQ/rasa: docs/docs/connectors/twilio.mdx

Restart your Rasa server
to make the new channel endpoint available for Twilio to send messages to.

> #### Receiving location data via twilio (whatsapp)
> 
> You can receive location data (WhatsApp Location) from user from this channel, you just need to do the following.
> 
> 1. Make an intent name locationData and two entities and slots, one for Latitude and one for Longitude respectively.

Reviewer: ```suggestion
### Receiving Location Data from Whatsapp with Twilio connector

This is how you can receive location data (WhatsApp Location) from a user on this channel:

1. Create an intent named "locationData" and define two entities and slots for Latitude and Longitude respectively.
```
```

```yaml-rasa title="domain.yml"
intents:
  - locationData

slots:
   Latitude:
     type: text
     mappings:
     - type: from_entity
       entity: Latitude

   Longitude:
     type: text
     mappings:
     - type: from_entity
       entity: Longitude

entities:
  - Latitude
  - Longitude

```

> 2. Once user send the location message it will Trigger the locationData intent and Slots will be set from entities (Note: no need for training data for entities, its being handled through the channel).

Reviewer: ```suggestion
2. When the user sends a location message, the locationData intent will be triggered and the slots will be set from the entities. Note that you do not need to provide training data for the entities as they are handled by the channel.
```

Restart your Rasa server
to make the new channel endpoint available for Twilio to send messages to.

#### Receiving location data via twilio (whatsapp)

You can receive location data from user from this channel you need to do the following.

1. Make an intent name locationData and two entities/slots one for Latitude and one for Longitude respectively.

```yaml-rasa title="domain.yml"
intents:
  - locationData
slots:
   Latitude:
     type: text
     mappings:

>      - type: custom

Author: Done

> slots:
>    Latitude:
>      type: text
>      mappings:
>      - type: custom
> 
>    Longitude:
>      type: text
>      mappings:
>      - type: custom

Reviewer: ```suggestion
slots:
   Latitude:
     type: text
     mappings:
     - type: from_entity
       entity: Latitude

   Longitude:
     type: text
     mappings:
     - type: from_entity
       entity: Longitude
```

You can also set the slots with the entity extraction and avoid using the Custom Action altogether

entities:
  - Latitude
  - Longitude

actions:
  - action_location_data
```

> 2. Once user send the location message it will Trigger the locationData intent you can later handle everything in custom actions.

Reviewer: I think this custom action adds complexity to your PR and the documentation. It isn't necessary either since slot mapping will set the slots. Do you think it'll be a good idea to remove it?

Author: yeah ill think remove it
